### PR TITLE
Fix(CI): Fix some type & CI Warn

### PR DESCRIPTION
**Pull request by nkokla**

### CODE REVIEW

File: src/components/Map/Map.tsx [hunk 1]
```diff
@@ -111,18 +111,18 @@ const formatBodyPopup = ({
               : ''
           }
           ${
-            energie_utilisee
+            energie_utilisee && typeof energie_utilisee === 'string'
```
Comment (desoindx) [inline src/components/Map/Map.tsx:114]: Why do you need to cast those ? 
is there any other cases possible ?

File: src/components/Map/Map.tsx [hunk 2]
```diff
@@ -278,16 +271,19 @@ export default function Map() {
...
-        console.info('zoom =>', map.current.getZoom());
+        console.info('zoom =>', map.current?.getZoom());
         logSoughtAddress();
       }
     });
 
     map.current.on('load', () => {
       setMapState('loaded');
 
+      if (!map.current) return;
```
Comment (desoindx) [inline src/components/Map/Map.tsx:285]: Same ?

File: src/components/Map/Map.tsx [hunk 3]
```diff
@@ -435,35 +444,42 @@ export default function Map() {
...
+            .addTo(map.current);
+          shouldUpdate = true;
+          return {
+            marker,
+            ...sAddress,
+          };
+        } else {
+          return sAddress;
+        }
+      }) || [];
```
Comment (desoindx) [inline src/components/Map/Map.tsx:468]: I don't get this || [] ?

**Comment by nkokla** [comment]
@desoindx : I have update this PR and extend for some CI Warning. You can review. :-)

**Review by desoindx — APPROVED** [review]

**Pull request merged**